SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,412 @@
+
+
+		return true;
+	}
+
+	DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FCheckMinimumWorkers, FAutomationTestBase*, Test, int, Expected);
+	bool FCheckMinimumWorkers::Update()
+	{
+		int Actual = Strat->GetMinimumRequiredWorkers();
+		Test->TestEqual(
+			FString::Printf(TEXT("Checking strategy for minimum required workers. Actual: %d, Expected: %d"), Actual, Expected),
+			Actual, Expected);
+		return true;
+	}
+
+	DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FCheckStratIsReady, FAutomationTestBase*, Test, bool, Expected);
+	bool FCheckStratIsReady::Update()
+	{
+		Test->TestEqual(
+			FString::Printf(TEXT("Checking strategy for minimum required workers. Actual: %d, Expected: %d"), Strat->IsReady(), Expected),
+			Strat->IsReady(), Expected);
+		return true;
+	}
+
+	DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FSpawnDefaultPawnAtLocation, FName, Handle,
+		FVector, Location);
+	bool FSpawnDefaultPawnAtLocation::Update()
+	{
+		FActorSpawnParameters SpawnParams;
+		SpawnParams.bNoFail = true;
+		SpawnParams.SpawnCollisionHandlingOverride = ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+
+		AActor* NewActor = TestWorld->SpawnActor<ADefaultPawn>(Location, FRotator::ZeroRotator, SpawnParams);
+		TestActors.Add(Handle, NewActor);
+
+		return true;
+	}
+
+	DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FSpawnSpecialPawnAtLocation,
+		FName, Handle, FVector, Location);
+	bool FSpawnSpecialPawnAtLocation::Update()
+	{
+		FActorSpawnParameters SpawnParams;
+		SpawnParams.bNoFail = true;
+		SpawnParams.SpawnCollisionHandlingOverride = ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+
+		AActor* NewActor = TestWorld->SpawnActor<ASpecialPawn>(Location, FRotator::ZeroRotator, SpawnParams);
+		TestActors.Add(Handle, NewActor);
+
+		return true;
+	}
+
+	DEFINE_LATENT_AUTOMATION_COMMAND_THREE_PARAMETER(FCheckActorsHaveSameAuth,
+		FAutomationTestBase*, Test, FName, FirstActorName, FName, SecondActorName);
+	bool FCheckActorsHaveSameAuth::Update()
+	{
+		VirtualWorkerId FirstActorAuth = Strat->WhoShouldHaveAuthority(*TestActors[FirstActorName]);
+		VirtualWorkerId SecondActorAuth = Strat->WhoShouldHaveAuthority(*TestActors[SecondActorName]);
+
+		Test->TestEqual(
+			FString::Printf(TEXT("Checking actors have the same auth. Actor1: %d, Actor2: %d"), FirstActorAuth, SecondActorAuth),
+			FirstActorAuth, SecondActorAuth);
+		return true;
+	}
+
+	DEFINE_LATENT_AUTOMATION_COMMAND_THREE_PARAMETER(FCheckActorsHaveDifferentAuth, FAutomationTestBase*, Test, FName, FirstActorName, FName, SecondActorName);
+	bool FCheckActorsHaveDifferentAuth::Update()
+	{
+		VirtualWorkerId FirstActorAuth = Strat->WhoShouldHaveAuthority(*TestActors[FirstActorName]);
+		VirtualWorkerId SecondActorAuth = Strat->WhoShouldHaveAuthority(*TestActors[SecondActorName]);
+
+		Test->TestNotEqual(
+			FString::Printf(TEXT("Checking actors have different auth. Actor1: %d, Actor2: %d"), FirstActorAuth, SecondActorAuth),
+			FirstActorAuth, SecondActorAuth);
+
+		return true;
+	}
+
```

Comment:
Could refactor this and above method into 1 while an extra argument saying whether you expect them to be equal?

Reply:
yep, done